### PR TITLE
Saml support sso

### PR DESCRIPTION
**Pull request by kasonde**
### What does it do?

This PR creates a route that accepts a POST request for SAML login to work with Strapi. 

## TL;DR

We need to add an additional route to [this file](https://github.com/strapi/strapi/blob/master/packages/strapi-admin/ee/middlewares/features-routes/routes.js) in the Strapi SSO code to allow SAML login. Instead of sending a `GET` request to the route:

```jsx
/connect/:provider
```

Services like OKTA and OneLogin, i.e SAML SSO login strategies send a `POST` request there. This results in a `405` error being thrown. `Method not allowed` . 

## Details

It turns out that the `passport-saml` auth strategy is sending a `POST` request to the route `/connect/:provider` instead of the expected `GET` request. This was causing all login attempts to fail, throwing a 405 error. The issue was tracked to this [routes.js](https://github.com/strapi/strapi/blob/master/packages/strapi-admin/ee/middlewares/features-routes/routes.js) file. After adding the code below to the array: 

```jsx
{
  method: 'POST',
  path: '/connect/:provider',
  handler: 'authentication.providerLogin',
}
```

the SSO feature worked again. 

### How to test it?

Attempt to perform login with a provider like OneLogin or Okta. This currently exists on Strapi v3.x.x and v4.x-next

### CODE REVIEW

**Review by Convly — COMMENTED** [review]
The code looks good to me, I'll still need to test it with a SAML provider later.

Also, is there any specific reason why we're merging this fix/feat in V4 and not in V3? From what I see, there is no breaking change

**Comment by kasonde** [comment]
Yes there is no breaking change. I was not sure about the v3 release schedule so I put it here instead. It's definitely something useful for v3 as it stemmed from an EE customer problem.

**Comment by Convly** [comment]
Let's target the next V3 release then :slightly_smiling_face:

**Comment by Convly** [comment]
Closing in favour of #10690
